### ISSUE TITLE
New option to disable some or all noncritical emails

### ISSUE THREAD

**Issue by Lantizia** [openmediavault/openmediavault#1778]
This is on a fresh install of openmediavault_7.0-32-amd64.iso with all updates ran immediately after.

Enabling email notifications seems like a very useful thing to do for preventative maintenance, as when you look at the user interface the list of events look important...

 - CPU usage
 - File systems
 - Load average
 - Memory usage 
 - Process monitoring
 - S.M.A.R.T.

Often people will enter an email address of either a shared mailbox or a mailing list so that multiple administrators can be told about critical events.

However I've noticed a whole bunch of emails have come in with messages that don't seem important and more importantly don't match any of the events listed above...

**"A reboot of the system is required to complete a package upgrade."**
_(whomever just upgraded the system likely knows this due to a web interface notification telling them, so is this email reminder needed?)_

**"your user account was used to log in to management-nas.company.lan (10.0.0.2) via a web browser."**
_(this gets sent every time, I'd understand if perhaps it was from an IP not seen before... but it's EVERY time)_

**"The configuration has been changed. You must apply the changes in order for them to take effect."**
_(whomever just changed something likely knows because of the big yellow banner, is an email needed?)_

And so on... I'm sure there are more.

Is there a full list of the possible email messages somewhere (like those 3 above) which fall outside of the scope of the 6 possible events that are listed on the user interface?

And is possible to add a new event type to turn some or all of them off?

**Comment by votdev** [comment]
It is possible to add functionality to make those emails configurable, but i don't want to do that because IMO they are important. If users do not want to have them, add a email app filter rule to redirect those emails directly into trash. That is the simplest solution to make everyone happy without the need to add additional code and to overload OMV with configuration options.

**Comment by Lantizia** [comment]
Presumably you wouldn't be able to easy put your finger one exactly how many different kinds of messages could be sent via email that don't fit into the 6 existing event categories?

And presumably you wouldn't take a pull request even if I wrote one?

**Comment by votdev** [comment]
The idea is actually that the user is forcibly made aware of this information without the option of deactivating these messages. For example, you can't turn off the notifications from Google that your account has been used from a previously unknown location.

I know that some users don't like this, but that's my philosophy and therefore OMV's philosophy. Filtering these messages with the mail program should not really be a problem for anyone.

**Comment by votdev** [comment]
> **"your user account was used to log in to management-nas.company.lan (10.0.0.2) via a web browser."**
> _(this gets sent every time, I'd understand if perhaps it was from an IP not seen before... but it's EVERY time)_

In that case your browser is clearing the cookie that is used the prevent sending this email.

**Comment by votdev** [comment]
> **"The configuration has been changed. You must apply the changes in order for them to take effect."**
> _(whomever just changed something likely knows because of the big yellow banner, is an email needed?)_

That is not true. Modules might get marked as dirty when security packages are installed unattended.

**Comment by votdev** [comment]
> **"A reboot of the system is required to complete a package upgrade."**
> _(whomever just upgraded the system likely knows this due to a web interface notification telling them, so is this email reminder needed?)_

This is incorrect. The file that indicates that a reboot is necessary can be set by every Debian package. If this package is installed unattended, the user does not see the notification in the UI.

**Comment by Lantizia** [comment]
- Google wouldn't e-mail me every time I login from a computer I've used before
- No cookies were cleared
- Presumably this only applies if your system is set to auto-upgrade packages which my instance doesn't
- Then the CLI would also tell the user a system reboot is required and this e-mail isn't needed either

If you don't get that user choice is a good thing, that's fine... then it's time to close this I guess?

**Comment by MadWalnut** [comment]
I also think that the option of disabling some emails would really benefit OMV. Especially daily apticron notifications can become really annoying since there is nothing critical about them (security updates are installed via unattended upgrades anyways). In case someone is also annoyed about this specific email type, [here is a workaround](https://forum.openmediavault.org/index.php?thread/50315-how-to-disable-cron-apt-notifications-when-packages-are-not-installed/&pageNo=2#post395313).

I believe not being able to disable certain email types will just lead to users disabling the email feature altogether and then missing actual important information (e.g. SMART). A filter rule in the mail program is not a viable workaround since this will still spam my phone if my mail program on my computer is not running (and mobile clients often do not support advanced filters - at least mine does not).

I can follow your point of forcing users towards good practices but I think one also has to acknowledge that many (most?) OMV users are home users and here priorities often lie elsewhere. I am not saying security is not important, but I really do not think shoving the (very) frequent updates of e.g. docker engine packages or OMV plugins in the users face every day makes for an enjoyable experience (again, security updates are already handled by unattended upgrades). At that point the home server just feels like a second day job. Just my two cents. You may of course disagree @votdev but I would also plead for at least some optional ENVs for advanced users (if GUI toggles are out of discussion). Especially since this was already possible in OMV6 and only removed for OMV7.

**Comment by votdev** [comment]
> A filter rule in the mail program is not a viable workaround since this will still spam my phone if my mail program on my computer is not running

Most email providers support filter rules in their webui.

> I would also plead for at least some optional ENVs for advanced users

Not necessary because all services support adding custom config files, e.g. `/etc/apt/listchanges.conf.d/` or `/etc/apt/apt.conf.d/`.

**Comment by Lantizia** [comment]
It's also kind of misrepresented, as with any new install of OMV, it won't send any of these email at all unless you enable it.

But when you enable them, you're not properly informed by the interface as to the complete list of things you're agreeing to actually start receiving.  You can see a list of what you **think** you'll be receiving, but it's an incomplete list, basically a list of things any user would think would be useful things to get e-mails about... not realising they're also going to get a tonne of emails that are just annoying.

**Comment by Gwindalmir** [comment]
All it does is train users to ignore all the emails.
Why add a filter for every unimportant message that comes it, when it's easier to 1-click block the entire sender? Or, for installations like mine, just disable the entire email system in OMV.

I fought for this feature a year or two ago, same result. I mentioned it then, OMV is user-hostile.

I'm literally the only user on the system, in a home network. There's no way (barring a more serious malware attack) someone else would be able to login.
I only care about SMART messages, but I have to agree to undocumented useless junk to get those.

**Comment by votdev** [comment]
> What's even funnier, is that the login emails are useless, because it always says 127.0.0.1 or ::1 as the IP address, no matter what system I log in from (probably due to the reverse-proxy). That's not helpful.

Fixed with https://github.com/openmediavault/openmediavault/commit/90406411ba8fcb2f04c4abbb923be51aaf35567d

**Comment by ghemberg** [comment]
I understand the importance of keeping a system up-to-date.

However, I feel that bugging admins with multiple mails every day to pressure them into taking action is not a good way to go. Because of this annoyance, I am seriously considering disabling all mails from OMV as...

- ... I don't have an overview of the different mails that could be sent out (perhaps System > Notification > Events could list all possible mails, even if some couldn't be disabled) and I don't want to manually fiddle with the system every time a new one pops up
- ... these mails are no different than spam for anyone managing several OMV instances.

I would personally be in favour of letting admins manage all mails they may or may not wish to receive...

**Comment by mjlehrke** [comment]
This spam is seriously annoying. There needs to be a way to disable these specific emails.

**Comment by gitchre** [comment]
Im sorry. This is just spam. After switching from OMV6 to 7 I'm just drowning in several emails a day bugging me to update minor packages. After reading this. Don't bother to change - you made your point clear - fair enough - probably everyone has just disabled the notfications already.

**Comment by ryecoaaron** [comment]
several emails is drowning??

Create the following saltstack file:

`/srv/salt/omv/deploy/postfix/99header_checks.sls`
```
append_header_checks_to_postfix:
  file.append:
    - name: /etc/postfix/main.cf
    - text: "header_checks = regexp:/etc/postfix/header_checks"
    - require:
      - file: configure_postfix_main
```
then run:
```
sudo /usr/bin/salt-call --local saltutil.clear_cache
sudo omv-salt stage run prepare
```

then people could add rules like the following to disable specific emails:
```
echo '/^Subject:.*package update(s) available/    DISCARD' | sudo tee -a /etc/postfix/header_checks
echo '/^Subject:.*Your user account was used to log in to the openmediavault control panel via a web browser/    DISCARD' | sudo tee -a /etc/postfix/header_checks
sudo postmap /etc/postfix/header_checks
sudo systemctl reload postfix
```

**Comment by eremem** [comment]
> In that case your browser is clearing the cookie that is used the prevent sending this email.

@votdev This doesn't seem to be the case for Firefox (with Chrome it seems to actually work so).
With Firefox the email is sent every time I log in having the same tab open and active.
I have no ad blocker active, privacy settings are set to their defaults. I'm at version 7.4.9-2 and observe this annoying behavior since about the latest 2-3 OMV updates.

**Comment by 1yhuag1** [comment]
> several emails is drowning??
> 
> Create the following saltstack file:
> 
> `/srv/salt/omv/deploy/postfix/99header_checks.sls`
> 
> ```
> append_header_checks_to_postfix:
>   file.append:
>     - name: /etc/postfix/main.cf
>     - text: "header_checks = regexp:/etc/postfix/header_checks"
>     - require:
>       - file: configure_postfix_main
> ```
> 
> then run:
> 
> ```
> sudo /usr/bin/salt-call --local saltutil.clear_cache
> sudo omv-salt stage run prepare
> ```
> 
> then people could add rules like the following to disable specific emails:
> 
> ```
> echo '/^Subject:.*package update(s) available/    DISCARD' | sudo tee -a /etc/postfix/header_checks
> echo '/^Subject:.*Your user account was used to log in to the openmediavault control panel via a web browser/    DISCARD' | sudo tee -a /etc/postfix/header_checks
> sudo postmap /etc/postfix/header_checks
> sudo systemctl reload postfix
> ```

我试了一下，很好用，但是“[openmediavault.local] 1 package update(s) available”还是没法拦截。
有人告诉我，这是因为：“/^Subject:.*package update(s) available/    DISCARD”里，“（s）”没有标记，被系统以为是正则表达式的缘故。。。
让我换成：“/^Subject:.*package update\(s\) available/ DISCARD”
今晚我测试一下可不可行。

**Comment by votdev** [comment]
Notifications sent by APT package management can be disabled now. See https://github.com/openmediavault/openmediavault/pull/1845

**Comment by Lantizia** [comment]
> This issue has been automatically marked as stale because it has not had any activity for 60 days. It will be closed if no further activity occurs for another 30 days.

lol

**Comment by Lantizia** [comment]
lol again

**Comment by ryecoaaron** [comment]
> Additionally, as helpdesk and SOC analyst, it's well known that too much reporting leads to alert fatigue, causing admins to miss alerts, ignore them, etc. So you are really creating the problem you are trying to prevent.

And yet alert fatigue is still a problem with almost every enterprise product.  This is why email rules/filters are so helpful.

> As a developer, it's advisable to give the options and make best practices recommendations, rather than deciding for the users. Most people are annoyed or even hate companies like Apple for this exact reason. Nobody likes being told what to do. 

Yet you are telling the dev what to do?  Try to remember you are dealing with a product developed by one person.  While you think it is easy to just give everyone what they want, that requires time to develop the changes and future time to maintain it.  If these changes are so desperately needed, the community should contribute.

**Comment by Gwindalmir** [comment]
> > Additionally, as helpdesk and SOC analyst, it's well known that too much reporting leads to alert fatigue, causing admins to miss alerts, ignore them, etc. So you are really creating the problem you are trying to prevent.
> 
> And yet alert fatigue is still a problem with almost every enterprise product. This is why email rules/filters are so helpful.
> 
As someone with ADHD, I have rules/filters, and I always forget to check the various folders I place the messages in. Not really important, just wanted to chime in that rules/filters aren't the end-all-be-all solution.

> > As a developer, it's advisable to give the options and make best practices recommendations, rather than deciding for the users. Most people are annoyed or even hate companies like Apple for this exact reason. Nobody likes being told what to do.
> 
> Yet you are telling the dev what to do? Try to remember you are dealing with a product developed by one person. While you think it is easy to just give everyone what they want, that requires time to develop the changes and future time to maintain it. If these changes are so desperately needed, the community should contribute.

The "needs of the many".

I honestly got so tired of it, I just disabled the entire notification system. I host the system for a family of 3 people, I don't need the level of stress/spam those notifications cause for that use case. I just check the dashboard occasionally for important flags.

**Comment by BadBunnyHimself** [comment]
I have had my OMV system running for several months now and decided to enable all the notification options in System > Notifications > Settings, and left all the default options in System > Notifications > Events ticked.
Oh hell, the damn system has not stopped sending emails now, about every 30 seconds I think for a few hours.
I disabled the notifications but they kept coming.
After a bit of Googling I found how to ssh into the server and used the option postqueue -p to see how many emails were queued up - over 550 of them!
I used postsuper -d ALL to remove them and checked again with postqueue -p and they were gone.
I hope this was OK to do, I was happy to not get new emails, however it didn't stop there, now i am getting hundreds of "Undelivered Mail Returned to Sender" messages. This is a nightmare!
I hope my email address is not going to get blacklisted because of this.

EDIT: Even though on the settings page for notifications I had unticked the box that said 'Enabled' I was still getting all the notifications. Being generally suspicious minded and distrusting by nature, I proceeded to clear all the settings fields of all my email information, and would you know it, all the notifications and bounced emails suddenly stopped.

**Comment by dexter74** [comment]
Just edit smtp url for force to edit file & stop mail.

**Comment by Lantizia** [comment]
When I opened this originally I listed that these options existed to enable/disable reasons to e-mail the user...

>     * CPU usage
> 
>     * File systems
> 
>     * Load average
> 
>     * Memory usage
> 
>     * Process monitoring
> 
>     * S.M.A.R.T.

At some point this was also added in the last year...

>     * Update Management

And now it sounds like #2003 adds something like this?

 >     * Authentication

So would you say this is a complete list now?  

For example, in my original post I talked about...

> **"The configuration has been changed. You must apply the changes in order for them to take effect."** _(whomever just changed something likely knows because of the big yellow banner, is an email needed?)_

Do we have a way to turn off these nag e-mails too?  

This issue wasn't logged because **some specific** notifications couldn't be disabled... 
...it was logged because **all unspecified noncritical** notifications that currently couldn't be disabled... couldn't be disabled :)

So this issue can't really be marked as resolved... or even really worked on... without **first** establishing what **all** the possibilities actually are for noncritical email notifications!

Sorry if that sounds pedantic... but I didn't log this issue just because of **some** specific e-mails (I just gave some examples).  Instead it was to get toggles for all "noncritical emails" (as per the title of the issue).

**Comment by Lantizia** [comment]
Most appreciated!  Thank you!
